File: src/DoAnTinHoc/GPUClass.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package DoAnTinHoc;

/**
 *
 * @author TRAN PHUC VAN
 */
public class GPUClass {
    public String manufacturer, productName, bus, memType, gpuChip;
    public int releaseYear, memSize, memBusWidth, gpuClock, memClock, unifiedShader, tmu, rop;
    public String pixelShader, vertexShader;
    public boolean igp;

    private boolean isNumeric(String str) {
        return str != null && !str.isEmpty() && str.matches("\\d+");
    }

    public GPUClass(String[] fields) {
        manufacturer = fields[0];
        productName = fields[1];
        releaseYear = isNumeric(fields[2]) ? Integer.parseInt(fields[2]) : 0;
        memSize = isNumeric(fields[3]) ? Integer.parseInt(fields[3]) : 0;
        memBusWidth = isNumeric(fields[4]) ? Integer.parseInt(fields[4]) : 0;
        gpuClock = isNumeric(fields[5]) ? Integer.parseInt(fields[5]) : 0;
        memClock = isNumeric(fields[6]) ? Integer.parseInt(fields[6]) : 0;
        unifiedShader = isNumeric(fields[7]) ? Integer.parseInt(fields[7]) : 0;
        tmu = isNumeric(fields[8]) ? Integer.parseInt(fields[8]) : 0;
        rop = isNumeric(fields[9]) ? Integer.parseInt(fields[9]) : 0;
        pixelShader = fields[10];
        vertexShader = fields[11];
        igp = fields[12].equalsIgnoreCase("Yes");
        bus = fields[13];
        memType = fields[14];
        gpuChip = fields[15];
    }

    public Object[] toRow() {
        return new Object[] {
            manufacturer, productName, releaseYear, memSize, memBusWidth,
            gpuClock, memClock, unifiedShader, tmu, rop, pixelShader,
            vertexShader, igp, bus, memType, gpuChip
        };
    }

    public String toCSV() {
        return String.join(",",
            manufacturer,
            productName,
            String.valueOf(releaseYear),
            String.valueOf(memSize),
            String.valueOf(memBusWidth),
            String.valueOf(gpuClock),
            String.valueOf(memClock),
            String.valueOf(unifiedShader),
            String.valueOf(tmu),
            String.valueOf(rop),
            String.valueOf(pixelShader),
            String.valueOf(vertexShader),
            igp ? "Yes" : "No",
            bus,
            memType,
            gpuChip
        );
    }
}

File: LinkedList/LinkedList/GPUClass.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LinkedList
{
    public class GPUClass
    {
        private string manufacturer, productName, bus, memType, gpuChip;
        private float releaseYear,memSize,memBusWidth,gpuClock,memClock,unifiedShader,tmu,rop,pixelShader,vertexShader;
        private bool igp;

        public GPUClass()
        {
            manufacturer = "";
            productName = "";
            bus = "";
            memType = "";
            gpuChip = "";
            releaseYear = 0;
            memSize = 0;
            memBusWidth = 0;
            gpuClock = 0;
            memClock = 0;
            unifiedShader = 0;
            tmu = 0;
            rop = 0;
            pixelShader = 0;
            vertexShader = 0;
            igp = false;
        } 

        public GPUClass(string manufacturer, string productName, string bus, string memType, string gpuChip, float releaseYear, float memSize, float memBusWidth, float gpuClock, float memClock, float unifiedShader, float tmu, float rop, float pixelShader, float vertexShader, bool igp)
        {
            this.manufacturer = manufacturer;
            this.productName = productName;
            this.bus = bus;
            this.memType = memType;
            this.gpuChip = gpuChip;
            this.releaseYear = releaseYear;
            this.memSize = memSize;
            this.memBusWidth = memBusWidth;
            this.gpuClock = gpuClock;
            this.memClock = memClock;
            this.unifiedShader = unifiedShader;
            this.tmu = tmu;
            this.rop = rop;
            this.pixelShader = pixelShader;
            this.vertexShader = vertexShader;
            this.igp = igp;
        }

        public string Manufacturer { get => manufacturer; set => manufacturer = value; }
        public string ProductName { get => productName; set => productName = value; }
        public string Bus { get => bus; set => bus = value; }
        public string MemType { get => memType; set => memType = value; }
        public string GpuChip { get => gpuChip; set => gpuChip = value; }
        public float ReleaseYear { get => releaseYear; set => releaseYear = value; }
        public float MemSize { get => memSize; set => memSize = value; }
        public float MemBusWidth { get => memBusWidth; set => memBusWidth = value; }
        public float GpuClock { get => gpuClock; set => gpuClock = value; }
        public float MemClock { get => memClock; set => memClock = value; }
        public float UnifiedShader { get => unifiedShader; set => unifiedShader = value; }
        public float Tmu { get => tmu; set => tmu = value; }
        public float Rop { get => rop; set => rop = value; }
        public float PixelShader { get => pixelShader; set => pixelShader = value; }
        public float VertexShader { get => vertexShader; set => vertexShader = value; }
        public bool Igp { get => igp; set => igp = value; }



    }
}
